Display rotation

jamwaffles opened the pull request:
Adds 0, 90, 180 and 270 degree rotations to the display. Usable like so:

```rust
let mut disp = Builder::new()
    .with_rotation(DisplayRotation::Rotate90)
    .connect_i2c(i2c);
```

Also adds the `get_dimensions()` method so consumers of this driver can position stuff relative to the display size.

therealprof commented on examples/rotation_i2c.rs:
```diff
@@ -0,0 +1,71 @@
+    let mut disp = Builder::new()
+        .with_rotation(DisplayRotation::Rotate90)
+        .connect_i2c(i2c);
+    disp.init();
```
I smell new warnings here. ;)

jamwaffles commented:
Thanks for reviewing. Fixed the example warnings and added a `set_rotation()` method. Annoyingly it has to return a `Result` because it sets some config in the display. Makes the API a little messy but it's necessary.

jamwaffles commented:
Boss!
